lib/modules/backupAndRestore/providers/GoogleDrive.js

```diff
@@ -28,9 +34,42 @@ export class GoogleDrive extends Provider {
 			await Permissions.request(['https://accounts.google.com/o/oauth2/v2/auth', 'https://content.googleapis.com/drive/v3/*']);
 		});
 
+		// Re-check email address cache, in case it got invalidated due to google account logout
+		if (!this.getCachedEmailAddress()) {
+			try {
+				this.emailAddress = await this.verifyToken();
+				this.setCachedEmailAddress(this.emailAddress);
+			} catch (err) {
+				console.error(err);
+			}
+		}
+
 		return this;
 	}
 
+	getCachedEmailAddress() {
+		const cachedEmailAddress = localStorage.getItem(CACHED_GOOGLE_EMAIL_ADDRESS);
+		return cachedEmailAddress || '';
+	}
+
+	setCachedEmailAddress(emailAddress: string) {
+		localStorage.setItem(CACHED_GOOGLE_EMAIL_ADDRESS, JSON.stringify(emailAddress));
+	}
+
+	async verifyToken(): Promise<string> {
+		const decryptedToken = await ajax({
+			method: 'GET',
+			url: 'https://www.googleapis.com/oauth2/v3/tokeninfo',
+			query: { access_token: this.accessToken },
+			headers: { 'Content-Type': 'application/json' },
```

Comment:
Redundant when `type: 'json'` is specified.

Reply:
Thanks, that makes sense. Should we also remove these redundant headers from other ajax calls where `type: 'json'` has already been specified, e.g. https://github.com/honestbleeps/Reddit-Enhancement-Suite/pull/4440/files/3ccdb7164d81aa4cc06fea43edd18be54c6c2c42#diff-0a606b28b4a58094611f298cd827e0b9R94 ?